src/test/scala/firrtlTests/ChirrtlSpec.scala

```diff
@@ -94,7 +94,37 @@ class ChirrtlSpec extends FirrtlFlatSpec {
     }
   }
 
-  it should "compile and run" in {
-    runFirrtlTest("ChirrtlMems", "/features")
+  "Chirrtl" should "disregard unused memory ports" in {
+    val input =
+     """circuit Queue :
+       |  module Queue :
+       |    input clock : Clock
+       |    cmem ram : UInt<1>[2]
+       |    infer mport T_107 = ram[UInt(0)], clock
+       """.stripMargin
+
+    object CheckQueue extends Pass {
+      val name = "Check Result"
+      def checkEmptyStmt(s: Statement): Boolean = s match {
+        case s: Block => s.stmts forall checkEmptyStmt
+        case EmptyStmt => true
+        case _ => false
+      }
+      def run(c: Circuit) = {
+        val errors = new Errors
+        val check = c.modules forall {
+          case m: Module => checkEmptyStmt(m.body)
+          case m: ExtModule => true
+        }
+        if (!check) {
+          errors append new PassException("Queue should be empty")
+          errors.trigger
+        }
+        c
+      }
+    }
+    (passes :+ CheckQueue).foldLeft(Parser.parse(input.split("\n").toIterator)) {
+      (c: Circuit, p: Pass) => p.run(c)
+    }
```

Comment:
This could also perhaps be more concisely described by writing a self-contained Firrtl Test (like features/ChirrtlMems.fir) and just check if it compiles or something via compileFirrtlTest.